external_search.py

```diff
@@ -882,8 +577,748 @@ def create_new_version(cls, search_client, base_index_name, version=None):
+
+        "football", "softball", "software", "postwar",
+
+        "tennis",
+
+        "hamlet", "harlem", "amulet", "tablet",
+
+        "biology", "ecology", "zoology", "geology",
+
+        "joke", "jokes" # "jake"
+
+        "cat", "cats",
+        "car", "cars",
+        "war", "wars",
+
+        "away", "stay",
+    ]
+
+    # If this regular expression matches a query, we will not run
+    # a fuzzy match against that query, because it's likely to be
+    # counterproductive.
+    #
+    # TODO: Instead of this, avoid the fuzzy query or weigh it much
+    # lower if there don't appear to be any misspelled words in the
+    # query string.
+    FUZZY_CIRCUIT_BREAKER = re.compile(
+        r'\b(%s)\b' % "|".join(FUZZY_CONFOUNDERS), re.I
+    )
+
+    def __init__(self, query_string, filter=None):
+        """Store a query string and filter.
+
+        :param query_string: A user typed this string into a search box.
+        :param filter: A Filter object representing the circumstances
+            of the search -- for example, maybe we are searching within
+            a specific lane.
+        """
+        self.query_string = query_string
+        self.filter = filter
+
+    def build(self):
+        """Make an Elasticsearch-DSL query object out of this query."""
+        query = self.query()
+
+        # Add the filter, if there is one.
+        if self.filter:
+            built_filter = self.filter.build()
+            if built_filter:
+                query = Q("filtered", query=query, filter=built_filter)
+
+        # There you go!
+        return query
+
+    def query(self):
+        """Build an Elasticsearch Query object for this query string.
+        """
+        query_string = self.query_string
+
+        # The search query will create a dis_max query, which tests a
+        # number of hypotheses about what the query string might
+        # 'really' mean. For each book, the highest-rated hypothesis
+        # will be assumed to be true, and the highest-rated titles
+        # overall will become the search results.
+        hypotheses = []
+
+        # Here are the hypotheses:
+
+        # The query string might appear in one of the standard
+        # searchable fields.
+        simple = self.simple_query_string_query(query_string)
+        self._hypothesize(hypotheses, simple)
+
+        # The query string might be a close match against title,
+        # author, or series.
+        self._hypothesize(
+            hypotheses,
+            self.minimal_stemming_query(query_string),
+            100
+        )
+
+        # The query string might be an exact match for title or
+        # author. Such a match would be boosted quite a lot.
+        self._hypothesize(
+            hypotheses,
+            self._match_phrase("title.standard", query_string), 200
+        )
+        self._hypothesize(
+            hypotheses,
+            self._match_phrase("author.standard", query_string), 200
```

Comment:
`title.standard` exists because we define it in the mapping (https://github.com/NYPL-Simplified/server_core/blob/b8023d3dc2f7399dd9b875593081799c8e086a8c/external_search.py#L502), but for `author` the standard analyzer is the default so you don't need the `.standard`, and actually I'm surprised if it works.

Reply:
I see this was in the old code too - I guess it does work.

Reply:
It wasn't working! Exact author matches weren't being boosted. That's why in `test_exact_matches`, a book whose title was "He Is Peter Graves" (i.e. "Peter Graves" plus some other stuff) was showing up before a book whose author was the literal string "Peter Graves".

I assumed that outcome was consistent with both results being boosted, but if that was the case the closer match should have come out on top.